client/my-sites/post-selector/selector.jsx

```diff
@@ -185,20 +200,27 @@ const PostSelectorPosts = React.createClass( {
 		return includes( requestedPages, lastPage ) && ! loading;
 	},
 
-	getItemHeight( item ) {
-		if ( item && this.itemHeights[ item.global_ID ] ) {
-			return this.itemHeights[ item.global_ID ];
+	getPostChildren( postId ) {
+		const { posts } = this.props;
+		return filter( posts, ( { parent } ) => parent && parent.ID === postId );
+	},
+
+	getItemHeight( item, _recurse = false ) {
+		if ( ! item ) {
+			return ITEM_HEIGHT;
 		}
 
-		let height = ITEM_HEIGHT;
+		if ( item.parent && ! _recurse && includes( this.postIds, item.parent.ID ) ) {
+			return 0;
+		}
 
-		if ( item && item.items ) {
-			height += item.items.reduce( ( memo, nestedItem ) => {
-				return memo + this.getItemHeight( nestedItem );
-			}, 0 );
+		if ( this.itemHeights[ item.global_ID ] ) {
```

Comment:
Confirmed that `setItemRef` is called on later `renderItem` (almost excessively so).